src/components/Modal/Modal.js

```diff
@@ -18,16 +20,18 @@ class Modal extends Component {
   }
 
   cloneWithProps = (child) => {
-    const name = child.type.displayName || child.type.name;
+    const { id, onClose } = this.props;
+    const type = child.type;
 
-    if (name === 'ModalContent') {
-      const { id } = this.props;
+    if (type === ModalContent) {
```

Comment:
@philippfrank you were right in your initial review. relying on displayName is not safe for functional components since fnNames get mangled by uglify in the default config